### PR TITLE
Give signature help for arbitrary functions called in .js files (even off of <dot>).

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

File: src/services/services.ts [hunk 1]
```diff
@@ -2408,15 +2452,8 @@ module ts {
...
-        /**
-         * Clean up any semantic caches that are not needed. 
-         * The host can call this method if it wants to jettison unused memory.
-         * We will just dump the typeChecker and recreate a new one. this should have the effect of destroying all the semantic caches.
-         */
         function cleanupSemanticCache(): void {
-            if (program) {
-                typeInfoResolver = program.getTypeChecker();
-            }
+            // TODO: Should we jettison the program (or it's type checker) here?
```
Comment (DanielRosenwasser) [inline src/services/services.ts]: its

**Comment by CyrusNajmabadi** [comment]
This lets us perform sig help even when you're invoking a method on a value you have no type information for.

**Comment by DanielRosenwasser** [comment]
I meant that specific commit, not the general change.

**Comment by JsonFreeman** [comment]
:+1:

**Pull request merged**